### PR TITLE
Connect GraphQL to Telescope

### LINKED ISSUE #697: Switch from REST to GraphQL for Gatsby to Telescope
**What would you like to be added**:

Let's use the [gatsby-source-graphql](https://www.gatsbyjs.org/packages/gatsby-source-graphql/) plugin to allow using our Telescope GraphQL API instead of getting posts/feeds/etc from the REST API.

**Why would you like this to be added**:

Gatsby uses GraphQL all the way through, so we might as well embrace that.  Also, we have the GraphQL endpoint on Telescope, so we might as well embrace that too!

### PR DESCRIPTION
**Pull request by c3ho**
## Issue This PR Addresses
Fixes #697 

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [x] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description
This is actually pretty confusing to use and I've only had success with getting the data as a StaticQuery instead of a Page.  To test changes, use `npm run develop` and navigate to http://localhost:8000/__graphql. 

There should now be a checkbox called `telescope` on the left hand side. Once you click on that you'll find all the other queries we have set up. The `telescope` is like a wrapper. The great thing is, you can build your query in there and then export it using `Code Exporter`. Make sure when you export you change the dropdown from `Page Query` to `StaticQuery`

image

One thing to watch out for is that when you get the data, its an object, so to get at the data that I want in my above query(title) I have to do something like `JSON.stringify(data.telescope.getPosts)`

## Checklist

- [ ] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

**Comment by cindyledev** [comment]
Wow this is some pretty cool stuff!
Annotation 2020-02-27 131142

**Comment by cindyledev** [comment]
Trying it with 'Ci'
Annotation 2020-02-27 131142_

Just 'C'
Annotation 2020-02-27 131142__

Amazing...

**Comment by c3ho** [comment]
Honestly I’m not too sure either, but if we were to use the plugin @humphd mentioned it’s for gatsby only

**Comment by manekenpix** [comment]
Yes, but what I mean is that I thought this PR would add queries or some changes that would fetch data from the backend to populate the frontend (and if possible, use filter capabilities).

File: src/frontend/gatsby-config.js [hunk 1]
```diff
@@ -49,5 +49,14 @@ module.exports = {
       },
     },
     `gatsby-theme-material-ui`,
+    {
+      resolve: `gatsby-source-graphql`,
+      options: {
+        typeName: `Telescope`,
+        fieldName: 'telescope',
+        url: 'https://dev.telescope.cdot.systems/graphql',
```
Comment (manekenpix) [inline src/frontend/gatsby-config.js]:
```suggestion
        url: `${telescopeUrl}/graphql`,
```

**Pull request merged**